pkg/controllers/user/rbac/roletemplate_handler.go

```diff
@@ -73,16 +81,109 @@ func (c *rtLifecycle) syncRT(template *v3.RoleTemplate, usedInProjects bool) err
 					return err
 				}
 				if len(verbs) > 0 {
-					_, err := c.m.reconcileRoleForProjectAccessToGlobalResource(resource, rt, verbs)
+					roleName, err := c.m.reconcileRoleForProjectAccessToGlobalResource(resource, rt, verbs)
 					if err != nil {
 						return errors.Wrapf(err, "couldn't reconcile role for project access to global resources")
 					}
+					rolesToKeep[roleName] = true
 				}
 			}
 		}
-
 	}
 
+	for _, obj := range prtbs {
+		prtb, ok := obj.(*v3.ProjectRoleTemplateBinding)
+		if !ok {
+			continue
+		}
+		rtbUID := string(prtb.UID)
+		set := labels.Set(map[string]string{rtbUID: owner})
+		crbs, err := c.m.crbLister.List("", set.AsSelector())
+		if err != nil {
+			return err
+		}
+		if parts := strings.SplitN(prtb.ProjectName, ":", 2); len(parts) == 2 && len(parts[1]) > 0 {
+			projectName := parts[1]
+			var roleVerb, roleSuffix string
+			for _, rtName := range template.RoleTemplateNames {
+				var createNS bool
+				rt, err := c.m.rtLister.Get("", rtName)
+				if err != nil {
+					return err
+				}
+				for _, rule := range rt.Rules {
+					if slice.ContainsString(rule.Resources, "namespaces") && len(rule.ResourceNames) == 0 {
+						if slice.ContainsString(rule.Verbs, "*") || slice.ContainsString(rule.Verbs, "create") {
+							roleVerb = "*"
+							createNS = true
+							break
+						}
+					}
+				}
+				if rt.Rules == nil {
+					// Kubernetes admin/edit roles
+					if rt.External && rt.Context == "project" && (rt.Name == "admin" || rt.Name == "edit") {
+						roleVerb = "*"
+						createNS = true
+					}
+				}
+
+				if roleVerb == "" {
+					roleVerb = "get"
+				}
+				roleSuffix = projectNSVerbToSuffix[roleVerb]
+				role := fmt.Sprintf(projectNSGetClusterRoleNameFmt, projectName, roleSuffix)
+				rolesToKeep[role] = true
+				if createNS {
+					rolesToKeep["create-ns"] = true
+				}
+				for resource := range globalResourcesNeededInProjects {
+					verbs, err := c.m.checkForGlobalResourceRules(rt, resource)
+					if err != nil {
+						return err
+					}
+					if len(verbs) > 0 {
+						roleName, err := c.m.reconcileRoleForProjectAccessToGlobalResource(resource, rt, verbs)
+						if err != nil {
+							return err
+						}
+						rolesToKeep[roleName] = true
+					}
+				}
+			}
+			for _, crb := range crbs {
+				if !rolesToKeep[crb.RoleRef.Name] {
+					if err := c.m.clusterRoleBindings.Delete(crb.Name, &metav1.DeleteOptions{}); err != nil {
+						return err
+					}
+				} else {
+					delete(rolesToKeep, crb.RoleRef.Name)
+				}
+			}
+			for role := range rolesToKeep {
+				subject, err := pkgrbac.BuildSubjectFromRTB(prtb)
+				if err != nil {
+					return err
+				}
+				_, err = c.m.clusterRoleBindings.Create(&rbacv1.ClusterRoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
```

Comment:
Is there a way to avoid the clusterRoleBindings.Create incase the crb already exists - if the crb is present in "crbs" maybe we can skip the create?